docs/IR.md

```diff
@@ -210,6 +211,8 @@ is explicitly included in the signature. The attribute_proto field describes att
 
 The opset_import and node fields describe the implementation of the function.
 
+The value_info field (added in IR version 10) allows a model to store type and shape information about the values used in a function, including its inputs and outputs. Note that this is optional, and ONNX allows functions to be polymorphic.
```

Comment:
When the value infos do not match the runtime type/shape when a function is invoked. Is the behavior defined? Should they be ignored?

Reply:
It's the same as for the main graph's value infos. I would characterize it as an error (so, no guaranteed behavior). In practice, this info can be used to perform optimizations (which may be incorrect, if incorrect value infos are supplied).

Reply:
Should we allow multiple signatures of the same function? Should we allow to define a function supported FLOAT, DOUBLE, FLOAT?

Reply:
> Should we allow multiple signatures of the same function? Should we allow to define a function supported FLOAT, DOUBLE, FLOAT?

Yes, that is useful. That is the subject of a different PR (https://github.com/onnx/onnx/pull/5011). The two are being done as separate PRs for various reasons,